### DISCUSSION TITLE
Cannot POST /shop_api

### DISCUSSION THREAD

**Discussion by wordxpression** [vendurehq/vendure#933] [Q&A] [answered]
Trying my first steps with connecting Next to Vendure and I get a problem with 'Cannot POST /shop_api'... 

Next server is running on localhost:8080 , the Vendure server on localhost:3000

Vendure admin client and GraphiQL are working fine...

Any ideas what I'm doing wrong?

**Comment by michaelbromley** [comment, marked as answer]
Next dev server also defaults to port 3000 no? Could it be a port conflict? When I was working on the Vendure/Next integration I would set the Vendure port to 3001 to avoid this.

**Reply by wordxpression** [reply]
Next is active on port 8080, so that can't be the problem.

**Reply by michaelbromley** [reply]
Oh yeah I should read more carefully next time 😆 
Ok, is the network call going to `http://localhost:3000/shop-api` or ` http://localhost:8080/shop-api`? Is there any useful error output by next.js?

**Reply by wordxpression** [reply]
http://localhost:3000/ shop-api
When I open the same URL in GraphiQL there are no problems...

**Reply by wordxpression** [reply]
For a moment I had the idea it could be a CORS problem, but I set the both  CORS options in the config file on every possible combination, but none of them solved the problem.

**Reply by michaelbromley** [reply]
Is there any other information in the error message or request? I'm not sure what the issue is and there is very little to go off.

**Reply by wordxpression** [reply]
The error message is outputted generated by the Vendure server. So I think I can assume, the message 'arrives' properly and it's not a CORS related problem. .. vendure knows the message arrives.

Do you have a suggestion where to start debugging in the Vendure code?

**Reply by michaelbromley** [reply]
Can you please screenshot the error in the vendure server console?

**Comment by wordxpression** [comment]
Sure. Attachment included
![cannot-post-vendure](https://user-images.githubusercontent.com/4310678/121884099-d1ece200-cd12-11eb-978f-de5a33f7f445.png)

**Reply by michaelbromley** [reply]
try replacing the underscore with a dash: `/shop-api` rather than `/shop_api`.

**Reply by wordxpression** [reply]
Ok. that's super stupid from my side. I completely overlooked it...

Going to try it right now, but I'm sure it is it. 

Sorry for wasting your time :(

**Reply by wordxpression** [reply]
That was indeed the problem.... solved.

So stupid I kept overlooking it 

Thanks for your patience 👍 :

**Reply by michaelbromley** [reply]
Don't worry about it. Stuff like that happens to everyone from time to time. Myself most _definitely_ included 😆

**Reply by wordxpression** [reply]
Ok ;) For me it was at least a good motivation to delve deeper in the code itself... I like the product...
